ui/page/send.go

```diff
@@ -0,0 +1,481 @@
+							})
+						}),
+					)
+				}),
+			)
+		},
+		func() {
+			layout.Flex{Axis: layout.Horizontal}.Layout(gtx,
+				layout.Rigid(func() {
+					pg.totalCostLabel.Layout(gtx)
+				}),
+				layout.Flexed(1, func() {
+					layout.Stack{Alignment: layout.NE}.Layout(gtx,
+						layout.Stacked(func() {
+							layout.Align(layout.Center).Layout(gtx, func() {
+								pg.totalCostValueLabel.Layout(gtx)
+							})
+						}),
+					)
+				}),
+			)
+
+			inset := layout.Inset{
+				Top: unit.Dp(30),
+			}
+			inset.Layout(gtx, func() {
+				layout.Flex{Axis: layout.Horizontal}.Layout(gtx,
+					layout.Rigid(func() {
+						pg.remainingBalanceLabel.Layout(gtx)
+					}),
+					layout.Flexed(1, func() {
+						layout.Stack{Alignment: layout.NE}.Layout(gtx,
+							layout.Stacked(func() {
+								// TODO get this value from wallet balance
+								// i.e wallet balance - total cost
+								pg.theme.Body2("4.37280441 DCR").Layout(gtx)
+							}),
+						)
+					}),
+				)
+			})
+		},
+		func() {
+			for pg.nextButton.Clicked(gtx) {
+				if pg.validate(true) {
+					pg.isShowingConfirmationModal = true
+				}
+			}
+			nextBtn := pg.theme.Button("Next")
+			if pg.validate(false) {
+				nextBtn.Background = helper.LightBlueColor
+			} else {
+				nextBtn.Background = helper.GrayColor
+			}
+			nextBtn.Layout(gtx, pg.nextButton)
+		},
+	}
+
+	pg.container.Layout(gtx, len(widgetFuncs), func(i int) {
+		layout.UniformInset(unit.Dp(10)).Layout(gtx, widgetFuncs[i])
+	})
+
+	if pg.isShowingConfirmationModal {
+		pg.drawConfirmationModal(gtx)
+	}
+
+	return
+}
+
+func (pg *Send) validate(setMessages bool) bool {
+	isValid := true
+
+	if pg.destinationAddressEditor.Text() == "" {
+		if setMessages {
+			pg.validationWidgets.destinationAddressLabel.Text = "Please enter a destination address"
+		}
+		isValid = false
+	} else {
+		// TODO check if destination address is a correct address for the current network
+		pg.validationWidgets.destinationAddressLabel.Text = ""
+	}
+
+	amount := pg.amountEditor.Text()
+	if amount == "" {
+		if setMessages {
+			pg.validationWidgets.amountLabel.Text = "Please enter an amount"
+		}
+		isValid = false
+	} else {
+		// check if amount is a valid number
+		_, err := strconv.ParseFloat(amount, 64)
+		if err != nil {
+			if setMessages {
+				pg.validationWidgets.amountLabel.Text = "Please enter a valid amount"
+			}
+			isValid = false
+		} else {
+			pg.validationWidgets.amountLabel.Text = ""
+		}
+	}
+
+	return isValid
+}
+
+func (pg *Send) drawConfirmationModal(gtx *layout.Context) {
+	modalWidgetFuncs := []func(){
+		func() {
+			layout.UniformInset(unit.Dp(10)).Layout(gtx, func() {
+				pg.modalWidgets.confirmLabel.Layout(gtx)
+			})
+		},
+		func() {
+			inset := layout.Inset{
+				Top:    unit.Dp(1),
+				Bottom: unit.Dp(1),
+			}
+			inset.Layout(gtx, func() {
+				pg.modalWidgets.line.Draw(gtx)
+			})
+		},
+		func() {
+			gtx.Constraints.Width.Min = gtx.Constraints.Width.Max
```

Comment:
Why?

Reply:
this is to make the following widget take up all available width

Reply:
Why not change the widgets size instead of changing the layout size?

Reply:
need the layout to take up all width, rather

Reply:
ok